lightning/src/routing/router.rs

```diff
@@ -284,7 +314,7 @@ impl Route {
 	/// Returns the total amount paid on this [`Route`], excluding the fees.
 	pub fn get_total_amount(&self) -> u64 {
 		return self.paths.iter()
-			.map(|path| path.split_last().map(|(hop, _)| hop.fee_msat).unwrap_or(0))
+			.map(|path| path.hops.split_last().map(|(hop, _)| hop.fee_msat).unwrap_or(0))
```

Comment:
Currently we include 1 path hop for the introduction node and 1 for the final node (or 1 hop total if they're the same). This mainly allows us to keep serialization of `PendingOutboundPayment::Retryable` intact + to not change the parts of the codebase where it accesses the path's last hop to get the path amount, iirc. I documented this behavior on `Path::hops` but maybe it could be made more clear. 

So, I think the blinded path fee is already subtracted since we're still splitting off the final destination hop, which still contains the full value of the payment.

Reply:
Sure, we can break `PendingOutboundPayment` out of the macro and write the blinded path, breaking backwards compat. 

If we're trying to contain all backwards compat breakage to 1 release, we may also want to consider breaking `PaymentPathFailed` events to contain the blinded portion. We have the option of maintaining backwards compat there using an odd TLV, however, though events read by past versions wouldn't have an accurate path if we did that.